elements/bulbs-dfp/bulbs-dfp.js

```diff
@@ -0,0 +1,128 @@
+import util from 'bulbs-elements/util';
+import invariant from 'invariant';
+import {
+  registerElement,
+  BulbsHTMLElement,
+} from 'bulbs-elements/register';
+
+// We have to do this little dance to properly subclass elements in Safari
+function BulbsHTMLDivElement () {}
+BulbsHTMLDivElement.prototype = HTMLDivElement.prototype;
+
+export default class BulbsDfp extends BulbsHTMLElement {
+  attachedCallback () {
+    invariant(this.hasAttribute('refresh-interval'),
+        '<div is="bulbs-dfp>" MUST specify a `refresh-interval` attribute in ms');
+
+    invariant(this.hasAttribute('viewport-threshold'),
+        '<div is="bulbs-dfp"> MUST specify a `viewport-threshold` property. ' +
+        'This value defines how many screens away from the viewport the slot ' +
+        'will be in order to trigger an ad load. 1.2 = 120% viewport height.');
+
+    util.getAnalyticsManager().sendEvent({
+      eventCategory: 'bulbs-dfp-element Metrics',
+      eventAction: 'attached',
+      eventLabel: this.dataset.adUnit,
```

Comment:
Emitting an attached event will let us track how many ads that are currently rendered on page load will be culled by the switch to `in-view` only ads.

When compared to the `enterviewport` metric, we'll know how many ads are rendered by user scrolling.